kore/src/Kore/Equation/Equation.hs

```diff
@@ -190,6 +191,13 @@ instance
             <> freeVariables right
             <> freeVariables ensures
 
+instance
+    AstWithLocation variable =>
+    AstWithLocation (Equation variable)
+  where
+    locationFromAst = locationFromAst . left
```

Comment:
The `requires` clause is actually the first pattern that appears in the Kore axiom:

```suggestion
    locationFromAst = locationFromAst . requires
```